ksql-engine/src/main/java/io/confluent/ksql/function/UdafFactoryInvoker.java

```diff
@@ -115,28 +107,4 @@ public boolean isVariadic() {
     return false;
   }
 
-  private static Object coerce(
-      final Class<?> clazz,
-      final String arg) {
-    if (Integer.class.isAssignableFrom(Primitives.wrap(clazz))) {
-      return Integer.valueOf(arg);
-    } else if (Long.class.isAssignableFrom(Primitives.wrap(clazz))) {
-      return Long.valueOf(arg);
-    } else if (Double.class.isAssignableFrom(Primitives.wrap(clazz))) {
-      return Double.valueOf(arg);
-    } else if (Float.class.isAssignableFrom(Primitives.wrap(clazz))) {
-      return Float.valueOf(arg);
-    } else if (Byte.class.isAssignableFrom(Primitives.wrap(clazz))) {
-      return Byte.valueOf(arg);
-    } else if (Short.class.isAssignableFrom(Primitives.wrap(clazz))) {
-      return Short.valueOf(arg);
-    } else if (Boolean.class.isAssignableFrom(Primitives.wrap(clazz))) {
-      return Boolean.valueOf(arg);
-    } else if (String.class.isAssignableFrom(clazz)) {
-      return arg;
-    } else if (Struct.class.isAssignableFrom(clazz)) {
```

Comment:
I wonder why we had a `Struct` here - was there ever a way to pass a struct? I don't see any tests for it, and I don't imagine that it's actually ever possible, but interesting to know the history of why this was added. The git blame have been munged recently, do you mind following that to see if there was anything in the original commit that explained why we added `Struct` here?

Reply:
This code is cut and paste from UdafTemplate.java and this line was added in this commit https://github.com/confluentinc/ksql/commit/c0ec37261e6de015996f30eb4e4306c7fc4595aa
I'm guessing because there was previous arg coercing code in there, you added the extra line to also coerce the struct, which seems a natural thing to do. However I don't think this code ever gets triggered.